R/rdeephaven/R/client_wrapper.R

```diff
@@ -1,110 +1,262 @@
-#' @title The Deephaven Client
-#' @description The Deephaven Client class is responsible for establishing and maintaining
-#' a connection to a running Deephaven server and facilitating basic server requests.
-#'
-#' @usage NULL
-#' @format NULL
-#' @docType class
-#'
-#' @examples
-#'
-#' # connect to the Deephaven server running on "localhost:10000" using anonymous 'default' authentication
-#' client_options <- ClientOptions$new()
-#' client <- Client$new(target = "localhost:10000", client_options = client_options)
-#'
-#' # open a table that already exists on the server
-#' new_table_handle1 <- client$open_table("table_on_the_server")
-#'
-#' # create a new dataframe, import onto the server, and retrieve a reference
-#' new_data_frame <- data.frame(matrix(rnorm(10 * 1000), nrow = 10))
-#' new_table_handle2 <- client$import_table(new_data_frame)
-#'
-#' # run a python script on the server (default client options specify a Python console)
-#' client$run_script("print([i for i in range(10)])")
-Client <- R6Class("Client",
-  public = list(
-
-    #' @description
-    #' Connect to a running Deephaven server.
-    #' @param target The address of the Deephaven server.
-    #' @param client_options ClientOptions instance with the parameters needed to connect to the server.
-    #' See ?ClientOptions for more information.
-    initialize = function(target, client_options) {
-      .verify_string("target", target)
-      if (class(client_options)[[1]] != "ClientOptions") {
-        stop(paste("'client_options' should be a Deephaven ClientOptions object. Got an object of type", class(client_options)[[1]], "instead."))
+#' @export
+setClass(
+  "Client",
+  representation(
+    .internal_rcpp_object = "Rcpp_INTERNAL_Client"
+  )
+)
+
+setGeneric(
+  "dhConnect",
+  function(target, ...) {
+    return(standardGeneric("dhConnect"))
+  },
+  signature = c("target")
+)
+
+#' @export
+setMethod(
+  "dhConnect",
+  signature = c(target = "character"),
+  function(target,
+           auth_type = "anonymous",
+           auth_token = "",
+           session_type = "python",
+           use_tls = FALSE,
+           tls_root_certs = "",
+           int_options = list(),
+           string_options = list(),
+           extra_headers = list()) {
```

Comment:
There is some design debate on if there should be `username` and `password` arguments for basic auth.  The answer will probably be yes.  For maximum flexibility, for basic auth, consider continuing to support `auth_token` as it is as well as `username` and `password`.  Just throw an error if they are both specified.

Reply:
Are we getting into a world where this signature is getting big and ugly, and could easily get bigger and uglier in the future? Is there still a reason that some R-idiomatic variant of `ClientOptions` is not more robust and flexible than this?